newrelic-weaver/src/test/java/com/newrelic/weave/weavepackage/WeavePackageAnnotationProcessingTest.java

```diff
@@ -0,0 +1,145 @@
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.Path");
+        expectedJaxRsMethodAnnotationClasses.add("javax.ws.rs.PATCH");
+
+        // Spring class annotations
+        expectedSpringAnnotationClasses.add("org.springframework.stereotype.Controller");
+        expectedSpringAnnotationClasses.add("org.springframework.web.bind.annotation.RestController");
+        // Spring method annotations
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.RequestMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.PatchMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.PutMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.GetMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.PostMapping");
+        expectedSpringMethodAnnotationClasses.add("org.springframework.web.bind.annotation.DeleteMapping");
+    }
+
+    @Test
+    public void testJaxRsAnnotationProcessing() throws IOException {
+        List<byte[]> weaveBytes = new ArrayList<>();
+        weaveBytes.add(WeaveTestUtils.getClassBytes(JAX_RS_SUBRESOURCE_ANNOTATIONS_CLASS_NAME));
+        weaveBytes.add(WeaveTestUtils.getClassBytes(JAX_RS_ANNOTATIONS_CLASS_NAME));
+        WeavePackageConfig config = WeavePackageConfig.builder().name("weave_unittest").source(
+                "com.newrelic.weave.weavepackage.testclasses").build();
+        final WeavePackage testPackage = new WeavePackage(config, weaveBytes);
+
+        Assert.assertEquals(expectedJaxRsAnnotationClasses, testPackage.getAllRequiredAnnotationClasses());
+        Assert.assertEquals(expectedJaxRsMethodAnnotationClasses, testPackage.getAllRequiredMethodAnnotationClasses());
+
+        final Map<String, ClassNode> allClassAnnotationWeaves = testPackage.getAllClassAnnotationWeaves();
+        final Map<String, ClassNode> allMethodAnnotationWeaves = testPackage.getAllMethodAnnotationWeaves();
+
+        // all class annotations should map to the JaxRsAnnotations ClassNode
+        for (String annotation : expectedJaxRsAnnotationClasses) {
+            String weaveNode = allClassAnnotationWeaves.get(annotation).name;
+            Assert.assertEquals(JAX_RS_ANNOTATIONS_CLASS_NODE_NAME, weaveNode);
+        }
+        // all method annotations should map to the JaxRsSubresourceAnnotations ClassNode, regardless of which JaxRs class bytes were processed first
+        for (String annotation : expectedJaxRsMethodAnnotationClasses) {
+            String weaveNode = allMethodAnnotationWeaves.get(annotation).name;
+            Assert.assertEquals(JAX_RS_SUBRESOURCE_ANNOTATIONS_CLASS_NODE_NAME, weaveNode);
+        }
+    }
+
+    @Test
+    public void testJaxRsAnnotationProcessingOppositeClassByteOrder() throws IOException {
```

Comment:
Could you explain how this test is different than the one above? I'm having trouble seeing the difference.

Reply:
```
        List<byte[]> weaveBytes = new ArrayList<>();
        weaveBytes.add(WeaveTestUtils.getClassBytes(JAX_RS_ANNOTATIONS_CLASS_NAME));
        weaveBytes.add(WeaveTestUtils.getClassBytes(JAX_RS_SUBRESOURCE_ANNOTATIONS_CLASS_NAME));
```

The two `add` lines are swapped which reverses the order that the `weaveBytes` are read in. This effectively mimics generating an instrumentation module jar and gives us control over the order that the classes are read from it.

Reply:
Thanks!  So same suggestion as above with the test name. 

`testJaxRsSubResourceAnnotationsProcessedLast`